veracruz-utils/src/csr.rs

```diff
@@ -67,27 +67,26 @@ pub fn generate_csr(private_key_der: &[u8]) -> Result<Vec<u8>> {
     Ok(csr)
 }
 
-pub fn generate_utc_time(
-    year: u32,
-    month: u32,
-    day: u32,
-    hour: u32,
-    minute: u32,
-    second: u32,
-) -> Result<Vec<u8>> {
-    if month > 11 || day > 30 || hour > 23 || minute > 59 || second > 59 {
-        return Err(anyhow!(CertError::InvalidUtcInputs {
-            month,
-            day,
-            hour,
-            minute,
-            second,
-        }));
-    }
-    let year = year % 2000;
-    let generated_time = format!(
-        "{:02}{:02}{:02}{:02}{:02}{:02}Z",
-        year, month, day, hour, minute, second
+#[cfg(features = "std")]
+pub fn generate_x509_time_now() -> [u8; 15] {
+    use chrono::{Datelike, Timelike, Utc};
+    use std::io::Write;
+
+    let mut buf = [0u8; 15];
+    let mut cursor = std::io::Cursor::new(&mut buf[..]);
+    let now = Utc::now();
+    let r = write!(
+        cursor,
+        "{:04}{:02}{:02}{:02}{:02}{:02}",
+        now.year(),
+        now.month(),
+        now.day(),
+        now.hour(),
+        now.minute(),
+        now.second()
     );
-    return Ok(generated_time.as_bytes().to_vec());
+    if r.is_err() || cursor.position() != 14 {
+        panic!("bad time")
```

Comment:
Is a panic the right thing to do here? or return an error?

Reply:
Panic is the right thing to do, I think. This can only fail if the year is negative, or greater than 9999, which could theoretically happen but I don't think we care about handling that nicely, or if the minutes are greater than 99, or something like that, which is an internal error.

Reply:
I think the code has "year 10000" problems in lots of places. I think the RFCs prescribe four-digit years.

Reply:
I think we can leave that problem for our supersmart descendants. If they aren't supersmart, then they will not be using computers by then, and they won't notice that we took this shortcut.